### PR TITLE
Improve exception handling

### PR DESCRIPTION
**Pull request by Azquelt**
Avoid leaking information when an exception is propegated up to the App server

Also use the correct Content-type when returning JSON error messages.

### CODE REVIEW

File: server/src/fat/java/com/ibm/ws/lars/rest/ApiTest.java [hunk 1]
```diff
@@ -1202,6 +1206,81 @@ public void test500Mapping() throws InvalidJsonAssetException, ParseException, I
...
+        assertThat(response, containsString("Test Error"));
+    }
+
+    /**
+     * Get a repository context that makes requests relative to the application root (rather than
+     * the LARS API root)
+     *
+     * @return a RepositoryContext relative to the application root
+     */
+    private RepositoryContext getRootContext() {
```
Comment (idlewis) [inline server/src/fat/java/com/ibm/ws/lars/rest/ApiTest.java]: Should this method be in the RepositoryContext class?

File: server/src/fat/java/com/ibm/ws/lars/rest/ApiTest.java [hunk 2]
```diff
@@ -1202,6 +1206,81 @@ public void test500Mapping() throws InvalidJsonAssetException, ParseException, I
...
+            throw new RuntimeException("Cannot create a root context based on " + repository.getFullURL());
+        }
+    }
+
+    /**
+     * Assert that a JSON error response is correct
+     */
+    private void assertJsonErrorResponse(int expectedStatusCode, String expectedMessage, String response) throws IOException {
+        try {
+            ObjectMapper mapper = new ObjectMapper();
```
Comment (idlewis) [inline server/src/fat/java/com/ibm/ws/lars/rest/ApiTest.java]: Mappers are quite expensive to create I believe, but are thread safe once constructed. Could this be a static?

File: server/src/fat/java/com/ibm/ws/lars/rest/ApiTest.java [hunk 3]
```diff
@@ -1202,6 +1206,81 @@ public void test500Mapping() throws InvalidJsonAssetException, ParseException, I
         mongoClient.close();
     }
 
+    @Test
+    public void testRuntimeException() throws ClientProtocolException, IOException {
```
Comment (idlewis) [inline server/src/fat/java/com/ibm/ws/lars/rest/ApiTest.java:1211]: Might be nice to have a comment in here to say where we are expecting the error to come from, i.e. this test is testing the error servlet, but the one below is testing the exception mapper in jax-rs..? I think ..??

File: server/src/fat/java/com/ibm/ws/lars/rest/ApiTest.java (hunk 1 above)
Comment (Azquelt) [reply server/src/fat/java/com/ibm/ws/lars/rest/ApiTest.java]: Put the test error servlet under the LARS API root so this method is no longer necessary and has been removed.

File: server/src/fat/java/com/ibm/ws/lars/rest/ApiTest.java (hunk 3 above)
Comment (Azquelt) [reply server/src/fat/java/com/ibm/ws/lars/rest/ApiTest.java:1211]: This is correct, I've moved the error servlet and given it a different name so it's easier to see that the tests are hitting different endpoints.

File: server/src/fat/java/com/ibm/ws/lars/rest/ApiTest.java (hunk 2 above)
Comment (Azquelt) [reply server/src/fat/java/com/ibm/ws/lars/rest/ApiTest.java]: Done

**Pull request merged**